packages/tenant-process/src/services/tenantService.ts

```diff
@@ -281,6 +283,41 @@ export function tenantServiceBuilder(
       });
     },
 
+    async deleteTenantMailById(
+      tenantId: TenantId,
+      mailId: string,
+      organizationId: TenantId,
+      correlationId: string
+    ): Promise<void> {
+      logger.info(`Deleting mail ${mailId} to Tenant ${tenantId}`);
+
+      if (tenantId !== organizationId) {
+        throw operationForbidden;
+      }
+
+      const tenant = await retrieveTenant(tenantId, readModelService);
+
+      if (!tenant.data.mails.find((m) => m.id === mailId)) {
+        throw mailNotFound(mailId);
+      }
+
+      const updatedTenant: Tenant = {
+        ...tenant.data,
+        mails: [...tenant.data.mails.filter((mail) => mail.id !== mailId)],
```

Comment:
No need to spread the array into a newly created one
```suggestion
        mails: tenant.data.mails.filter((mail) => mail.id !== mailId),
```